File: 2st/Yeonwoo/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = [0,0]
    x, y = (board[0]-1)//2, (board[1]-1)//2
    for i in keyinput:
        if i=="left" and answer[0]>-x:
            answer[0]-=1
        elif i=="right" and answer[0]<x:
            answer[0]+=1
        elif i=="up" and answer[1]<y:
            answer[1]+=1
        elif i=="down" and answer[1]>-y:
            answer[1]-=1
    return answer

File: 3st/Juhee/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = {0, 0};
        
        for(int i = 0; i < keyinput.length; i++) {
            if(keyinput[i].equals("left")) {
                answer[0]--;
                if(answer[0] < -(int)(board[0] / 2))
                    answer[0] = -(int)(board[0] / 2);
            } else if (keyinput[i].equals("right")) {
                answer[0]++;
                if(answer[0] > (int)(board[0] / 2))
                    answer[0] = (int)(board[0] / 2);
            } else if (keyinput[i].equals("up")) {
                answer[1]++;
                if(answer[1] > (int)(board[1] / 2))
                    answer[1] = (int)(board[1] / 2);
            } else {
                answer[1]--;
                if(answer[1] < -(int)(board[1] / 2))
                    answer[1] = -(int)(board[1] / 2);
            } 
        }
        
        return answer;
    }
}
